tests/autorag_research/nodes/retrieval/test_bm25.py

```diff
@@ -0,0 +1,235 @@
+
+    @patch("autorag_research.nodes.retrieval.bm25.LuceneSearcher")
+    def test_run_uses_contents_when_raw_not_available(self, mock_lucene_class):
+        """Should use contents attribute when raw is not available."""
+        mock_searcher = MagicMock()
+        mock_searcher.search.return_value = [
+            MockHit("doc1", 2.5, contents="Document content via contents"),
+        ]
+        mock_lucene_class.return_value = mock_searcher
+
+        module = BM25Module(index_path="/path/to/index")
+        results = module.run(["test query"], top_k=1)
+
+        assert results[0][0]["content"] == "Document content via contents"
+
+    @patch("autorag_research.nodes.retrieval.bm25.LuceneSearcher")
+    def test_run_empty_content_when_neither_raw_nor_contents(self, mock_lucene_class):
+        """Should return empty content when neither raw nor contents is available."""
+        mock_hit = MagicMock()
+        mock_hit.docid = "doc1"
+        mock_hit.score = 2.5
+        del mock_hit.raw
+        del mock_hit.contents
+
+        mock_searcher = MagicMock()
+        mock_searcher.search.return_value = [mock_hit]
+        mock_lucene_class.return_value = mock_searcher
+
+        module = BM25Module(index_path="/path/to/index")
+        results = module.run(["test query"], top_k=1)
+
+        assert results[0][0]["content"] == ""
+
+    @patch("autorag_research.nodes.retrieval.bm25.LuceneSearcher")
+    def test_run_no_results(self, mock_lucene_class):
+        """Should handle queries with no results."""
+        mock_searcher = MagicMock()
+        mock_searcher.search.return_value = []
+        mock_lucene_class.return_value = mock_searcher
+
+        module = BM25Module(index_path="/path/to/index")
+        results = module.run(["query with no results"], top_k=10)
+
+        assert len(results) == 1
+        assert results[0] == []
+
+    @patch("autorag_research.nodes.retrieval.bm25.LuceneSearcher")
+    def test_run_default_top_k(self, mock_lucene_class, mock_searcher):
+        """Should use default top_k of 10."""
+        mock_lucene_class.return_value = mock_searcher
+
+        module = BM25Module(index_path="/path/to/index")
+        module.run(["test query"])
+
+        mock_searcher.search.assert_called_with("test query", k=10)
+
+    @patch("autorag_research.nodes.retrieval.bm25.LuceneSearcher")
+    def test_run_result_structure(self, mock_lucene_class, mock_searcher):
+        """Should return results with correct structure."""
+        mock_lucene_class.return_value = mock_searcher
+
+        module = BM25Module(index_path="/path/to/index")
+        results = module.run(["test query"], top_k=3)
+
+        for result in results[0]:
+            assert "doc_id" in result
+            assert "score" in result
+            assert "content" in result
+            assert isinstance(result["doc_id"], str)
+            assert isinstance(result["score"], float)
+            assert isinstance(result["content"], str)
```

Comment:
So many tests. We need compact test codes than this. Since it uses the mock, it is just checking for single if-else statements in the function. Why do we need so many test codes for a single if-else statement and simple logic inside the module? These are also not good in my opinion. AI always makes test codes look beautiful and not compact, and I really hate it.